config/confighttp/confighttp_test.go

```diff
@@ -775,15 +775,15 @@ func TestServerAuth(t *testing.T) {
 			AuthenticatorID: config.NewComponentID("mock"),
 		},
 	}
+
 	host := &mockHost{
 		ext: map[config.ComponentID]component.Extension{
-			config.NewComponentID("mock"): &configauth.MockServerAuthenticator{
```

Comment:
This is a good example of how consumers would be able to use this. I like the new approach, especially as it allows consumers to omit specifying the default implementations to use, like demonstrated by the removal of line 785.